plugins/blackout-regex/test_blackout_regex.py

```diff
@@ -0,0 +1,314 @@
+        "group": None,
+        "duration": 3600,
+        "id": "1",
+    },
+    {
+        "status": "active",
+        "environment": "test",
+        "tags": [],
+        "service": ["service-([a-zA-Z])"],
+        "resource": None,
+        "event": None,
+        "group": None,
+        "duration": 3600,
+        "id": "2",
+    },
+    {
+        "status": "active",
+        "environment": "test",
+        "tags": ["site=site.*", "role=router"],
+        "service": [],
+        "resource": None,
+        "event": None,
+        "group": None,
+        "duration": 3600,
+        "id": "3",
+    },
+    {
+        "status": "active",
+        "environment": "test",
+        "tags": ["site=site.*", "role=firewall"],
+        "service": [],
+        "resource": None,
+        "event": None,
+        "group": None,
+        "duration": 3600,
+        "id": "4",
+    },
+    {
+        "status": "closed",
+        "environment": "test",
+        "tags": ["site=site.*", "role=router"],
+        "service": [],
+        "resource": None,
+        "event": None,
+        "group": None,
+        "duration": 3600,
+        "endTime": "1985-03-05T22:45:27.425Z",
+        "id": "5",
+    },
+    {
+        "status": "closed",
+        "environment": r"(rgx|env)",
+        "tags": [],
+        "service": ["test-.*"],
+        "resource": "FPC.*",
+        "event": "FPCDown",
+        "group": None,
+        "duration": 3600,
+        "endTime": "1985-03-05T22:45:27.425Z",
+        "id": "6",
+    },
+    {
+        "status": "active",
+        "environment": r"(rgx|env)",
+        "tags": [],
+        "service": [],
+        "resource": None,
+        "event": None,
+        "group": None,
+        "duration": 3600,
+        "id": "7",
+    },
+]
+
+
+class Model:
+    def __init__(self, **kwargs):
+        for arg, val in kwargs.items():
+            setattr(self, arg, val)
+
+    def __str__(self):
+        return "{name}({attrs})".format(
+            name=self.__class__.__name__,
+            attrs=", ".join(
+                [
+                    "{}={}".format(attr, getattr(self, attr))
+                    for attr in dir(self)
+                    if not attr.startswith("__")
+                ]
+            ),
+        )
+
+
+class Blackout(Model):
+    pass
+
+
+class Alert(Model):
+    def tag(self, tags):
+        self.tags.extend(tags)
+
+    def untag(self, tags):
+        for tag in tags:
+            if tag in self.tags:
+                self.tags.remove(tag)
+
+    def set_status(self, status):
+        self.status = status
+
+
+def _get_blackouts(**kwargs):
+    return ([Blackout(**blackout) for blackout in BLACKOUTS])
+    # return BLACKOUTS
+
+app.db.get_blackouts = _get_blackouts
+
+plugins_mod = sys.modules["alerta.plugins"] = MagicMock()
+plugins_mod.PluginBase = MagicMock
+
+
+from blackout_regex import BlackoutRegex  # pylama: ignore=E402
+
+log = logging.getLogger(__name__)
+
+
+class TestEnhance(unittest.TestCase):
```

Comment:
These tests seem to be a lot of duplicated code.  

Could you use something like https://docs.pytest.org/en/6.2.x/parametrize.html?